en/lessons/basics/operators.md

```diff
@@ -0,0 +1,164 @@
+operators, and how infix and prefix syntax can be interchanged.
+
+{% include toc.html %}
+
+
+## Using Operators
+
+Operators in Haskell are functions that can be placed between two arguments: they can
+return a value of any type and their names are either a string of symbols or ordinary
+function names surrounded in backticks. Using infix notation makes it easier to write
+expressions containing many common functions:
+
+```console?lang=haskell&prompt=ghci>,ghci|
+ghci> abs . subtract 930233356 $ 2 + 2
+930233352
+ghci> (`mod` 1119) . negate $ 4
+1115
+ghci> - (7 + 57843)
+-57850
+```
+__Note__: negate (`-`) is the only unary operator in Haskell.
+
+You may have seen operators such as `+`, `*`, and `.` before. The `$` operator is
+useful for removing brackets in function application: `abs . subtract 1 $ 2 + 2` is
+equivalent to `(abs . subtract 1) (2 + 2)`.
+
+### Prefix and Infix Notation
+
+To use an operator symbol in prefix notation, as if it were a regular function,
+surround it in brackets. Furthermore, you can create partially applied operators by
+putting an expression on only one side of the operator:
+
+```console?lang=haskell&prompt=ghci>,ghci|
+ghci> (+) 2 3
+5
+ghci> (5 +) . (* 2) $ 8
+21
+```
+
+Conversely, to make a regular function identifier an operator, surround it in backticks:
+
+```console?lang=haskell&prompt=ghci>,ghci|
+ghci> 2010 `mod` 1998
+12
+ghci> (`div` 3) . (* 1471684) $ 5
+2452806
+```
+
+
+### Operator Precedence and Associativity
+
+An operator's precedence and associativity combined is called its fixity, and it
+controls how the operator is parsed in an expression. To find out the fixity of an
+operator, look through `ghci`'s `:info` or the documentation for the operator:
+if fixity information isn't displayed, it defaults to `infixl 9`. The same applies to
+regular functions used in infix mode.
+```console?lang=haskell&prompt=ghci>,ghci|
+ghci> :info (+)
+infixl 6 +
+ghci> :i ($)
+infixr 0 $
+ghci> :i (==)
+infix 4 ==
+ghci> :i elem
+infix 4 `elem`
+```
+
+In Haskell, operators have a precedence level between 0 and 9: this precedence level
+is the number in the operator's fixity. To see how a group of expressions
+separated by operators would be bracketed, surround the highest precedence operator
+and the expressions next to it in brackets, and repeat this process until there
+is only one operator left at the highest level of the expression. Here's an example of
+applying this process:
+```
+16425973 + 17149685 * (106 + 235) == 3530784511
+         |          │              ╰─ Precedence level 4
+         |          ╰─ Precedence level 7
+         ╰─ Precedence level 6
+16425973 + (17149685 * (106 + 235)) == 3530784511
+(16425973 + (17149685 * (106 + 235))) == 3530784511
+```
+
+Knowing operator precedence is helpful in writing code without unnecessary brackets:
+
+```console?lang=haskell&prompt=ghci>,ghci|
+ghci> (^ (3 - 1)) . subtract 3818 $ 4
+14546596
+ghci> (1729 + 1) * 121 `div` 2
```

Comment:
Ensure that numbers are used in a way that makes the order of operations easy to deduce. For example here, the fact that the output is odd indicates that divides is of lower precedence than multiplication.